### ISSUE TITLE
Accessing firefox always shows adnade

### ISSUE THREAD

**Issue by copystring** [engageub/InternetIncome#59]
Hi,

I'm using this script with 10 proxies with a few services enabled. Two of them are eBesucher and Adnade. They are running fine, but when I set up an ssh tunnel to check the web interface, it will always show Adnade and never eBesucher. Feels like something is wrong with the setup of the containers.
Is it meant to be setup like this? If you ask me, the problem is port 300X and 400X are both redirected to port 5800 obviously opening the same Firefox instance. If Adnade and eBesucher are both running at the same time, surely for one of them the port has to be a different once. 5801 or so.
![image](https://github.com/engageub/InternetIncome/assets/1298861/c71c5242-acbd-4cd1-a9b2-34210b6c15ce)

**Comment by engageub** [comment]
Hello,
Thank you for raising this concern. Yes, you cannot start both in the same folder. Create a different folder to start adnade or ebesucher. 
Only one of them works at the same time. It is internal port of firefox container which we don't have control over. 
Do you see earnings for both adnade and ebesucher or only one ?

Thank you

**Comment by copystring** [comment]
I see earnings on both.

**Comment by engageub** [comment]
Thank you for the confirmation. Please update the code now to see both adnade and ebesucher containers.
Let me know if you see both the containers now. 

Thank you

**Comment by copystring** [comment]
It's working fine. Thank you!

**Comment by engageub** [comment]
Thank you for the confirmation. Closing this. Feel free to reopen in case you still any issues with this.

**Comment by copystring** [comment]
> I see earnings on both.

I have to correct my self. I've seen earnings for both on the same day while playing around with the script, but ever since I just let it run, I have not yet seen a single earning for adnade while ebesucher is running.

**Comment by engageub** [comment]
For adnade, if you stop and start from the same ip address, you have to wait for 5 minutes before running it.
What do you see inside the browser.
Thank you

**Comment by engageub** [comment]
For adnade and ebesucher, chrome versions are present in test branch.
They have recaptcha solvers included. Also for adnade, surfbar is present which gives 4X earnings and earnings are updated every minute in the Account statement.

https://github.com/engageub/InternetIncome/tree/test

Thank you

**Comment by copystring** [comment]
> For adnade, if you stop and start from the same ip address, you have to wait for 5 minutes before running it. What do you see inside the browser. If you are talking about the old version, that should be fine since you were able to see adnade in the browser. But let me know if there is any issue with the latest version. Thank you

I have not touched the docker containers since I last updated a week ago. Still no earnings.



> For adnade and ebesucher, chrome versions are present in test branch. They have recaptcha solvers included. Also for adnade, surfbar is present which gives 4X earnings and earnings are updated every minute in the Account statement.
> 
> https://github.com/engageub/InternetIncome/tree/test
> 
> Thank you

Yes, I want to test the test branch if you update it from main, so it contains the latest fixes.

**Comment by engageub** [comment]
Adnade containers are not restarted like ebesucher containers in main branch. May be that's one of the reason you are not seeing earnings.
But you have to look into the browser and take a screenshot. If you see any credits being added or some other message please update here.

Regarding test branch the steps to download are present in test branch Readme File. 
It is working fine in test branch also. It has more options than the  main branch. 

Thank you

**Comment by copystring** [comment]
Here is a Screenshot ![image](https://github.com/engageub/InternetIncome/assets/1298861/316f6e8b-11db-43b4-b7b6-37ec53c184f1)

Does the test branch include these two fixes? https://github.com/engageub/InternetIncome/commit/7b434354f0c252daee8230f0acaa0b878fcfbe74 and https://github.com/engageub/InternetIncome/commit/27764b064869cf805f9642ebac68cdaeeb08db87 I need them and this is why I did not test the test branch yet.

**Comment by engageub** [comment]
Thank you for the screenshot. The credits are visible, but "invalid access" is shown at the bottom. So these points are not being added to your adnade earnings ?
Please contact adnade support team to understand why this happens.
Regarding the test branch, those changes are already present in the branch.

Thank you

**Comment by copystring** [comment]
I'm loading the test branch. It shows `Ebesucher username is not configured. Ignoring Ebesucher..` but ebesucher username is set up correctly...

**Comment by copystring** [comment]
Odd. It still correctly add ebesucher after displaying the message from above.

**Comment by copystring** [comment]
This is how the ports are set up after installing the test branch: 
![image](https://github.com/engageub/InternetIncome/assets/1298861/1b7de94c-d623-4008-81eb-45eaa17131d4)
Is this correctly configured? Because I can not access any 400X port anymore with the same ssh tunnel as before.

**Comment by engageub** [comment]
> I'm loading the test branch. It shows `Ebesucher username is not configured. Ignoring Ebesucher..` but ebesucher username is set up correctly...

Ebesucher has both firefox and chrome. So may be for one of them it appeared. Messaging in the logs needs to be added whether it is for firefox or chrome. It is working fine.

**Comment by engageub** [comment]
> This is how the ports are set up after installing the test branch: ![image](https://user-images.githubusercontent.com/1298861/277044896-1b7de94c-d623-4008-81eb-45eaa17131d4.png) Is this correctly configured? Because I can not access any 400X port anymore with the same ssh tunnel as before.

From the screenshot, it looks like you are using chrome for adnade and firefox for ebesucher.
If you want to use firefox, you have to set the variable EBESUCHER_USE_CHROME=false in properties.conf file.

Thank you

**Comment by copystring** [comment]
> > This is how the ports are set up after installing the test branch: ![image](https://user-images.githubusercontent.com/1298861/277044896-1b7de94c-d623-4008-81eb-45eaa17131d4.png) Is this correctly configured? Because I can not access any 400X port anymore with the same ssh tunnel as before.
> 
> From the screenshot, it looks like you are using firefox for adnade and chrome for ebesucher.
> If you want to use chrome for adnade, you have to set the variable ADNADE_USE_CHROME=true in properties.conf file.
> It's also strange how you got port numbers for ebesucher firefox to adnade external  port numbers. 
> 
> Did you set anything to false in the code for ebesucher chrome or left it blank.
> 
> Thank you

Chrome is fine. I'd prefer to use chrome. But with chrome access to vnc adnade does not work. Vnc for ebesucher works.

**Comment by engageub** [comment]
Hi,
Could you please let me know how you were accessing the containers. 
Have you followed the steps mentioned in the following link to access them.

https://github.com/engageub/InternetIncome#thinking-how-to-access-127001-urls-for-ebesucher-or-mystnodes-if-gui-is-not-available-in-linuxquestion 

Thank you

**Comment by copystring** [comment]
I'm already using ssh tunnel just like before. I have not changed my ssh tunnel, and it worked 100% fine for ebesucher and adnade with the main branch. In the test branch, the same ssh tunnel does not work for ebesucher but works fine for adnade.

**Comment by engageub** [comment]
Hello,
It looks like you were using older version from test branch. You have to delete test.zip and download again. 
You are actually using firefox for one of them and not chrome.


Thank you

**Comment by engageub** [comment]
There are 2 issues.
1. You are using old version from test branch since the internal port of latest adnade is 3500
2. Running both with chrome, is causing internal port conflict for Xserver port 6900 for which I do not see option to change them as of now. If you are seeing earnings for adnade, you can continue using them. Else use ebesucher for firefox and adnade for chrome. The default settings in properties.conf have been updated. I will follow up with docker chromium team to see if there is an option to change Xserver port number.

**Comment by copystring** [comment]
OK. I downloaded the latest version again. Chrome for ebesucher and Firefox for adnade is OK? Works for me fine. Also, VNC via SSH tunnel for adnade and ebesucher works fine with this setup.

**Comment by engageub** [comment]
Yes, the vnc works fine as long as both are not chrome. But use chrome for adnade as you get 4x earnings. If you don't see recaptcha being asked for ebesucher, you may use firefox for ebesucher. 
If you see earnings for both ebesucher and adnade in chrome you may use it without vnc till next updates are made.  


Thank you

**Comment by copystring** [comment]
OK. Thank you. I need to use chrome for ebesucher because of captcha. I will wait until next update.

**Comment by engageub** [comment]
Recaptcha Extension is supported for firefox now. It will be updated soon to test branch, then main branch.

However if you wish to use chrome now for both adnade and ebesucher. Please download InternetIncome script in different folders and run them separately. 

Thank you

**Comment by engageub** [comment]
Hi,
Firefox in test branch has been updated with recaptcha solver. You can now use ebesucher with firefox in test branch. 


Thank you

**Comment by engageub** [comment]
Firefox in main branch now has recaptcha solver.

**Comment by copystring** [comment]
Thank you. I will test using the test branch and report back later or in a few days when earning for adnade and ebesucher work simultaneously

**Comment by copystring** [comment]
This is working pretty good. No issue so far! Thank you!
